### PR TITLE
Add Missing Navigation Buttons

### PR DESCRIPTION
**Pull request by tobitech**
Story: https://app.shortcut.com/smileid/story/11514/

## Summary

A few sentences/bullet points about the changes

## Known Issues

Any shortcomings in your work. This may include corner cases not correctly handled or issues related
to but not within the scope of your PR. Design compromises should be discussed here if they were not
already discussed above.

## Test Instructions

Concise test instructions on how to verify that your feature works as intended. This should include
changes to the development environment (if applicable) and all commands needed to run your work.

## Screenshot

If applicable (e.g. UI changes), add screenshots to help explain your work.

### CODE REVIEW

File: Example/SmileID/BiometricKYC/BiometricKycWithIdInputScreen.swift [hunk 1]
```diff
@@ -9,75 +9,81 @@ struct BiometricKycWithIdInputScreen: View {
...
-                            )
-                        }
+        ZStack {
+            switch viewModel.step {
+            case let .loading(messageKey):
+                VStack {
+                    ActivityIndicator(isAnimating: true).padding()
+                    Text(SmileIDResourcesHelper.localizedString(for: messageKey))
+                        .font(SmileID.theme.body)
+                        .foregroundColor(SmileID.theme.onLight)
```
Comment (jumaallan) [inline Example/SmileID/BiometricKYC/BiometricKycWithIdInputScreen.swift:19]: should we replace this with the .light option or this does not get affected by dark mode on the device

File: Example/SmileID/Helpers/CancellableNavigationView.swift [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    var body: some View {
+        NavigationView {
+            content()
+                .toolbar {
+                    ToolbarItem(placement: .cancellationAction) {
+                        Button {
+                            onCancel()
+                        } label: {
+                            Text(SmileIDResourcesHelper.localizedString(for: "Action.Cancel"))
+                                .foregroundColor(SmileID.theme.accent)
```
Comment (jumaallan) [inline Example/SmileID/Helpers/CancellableNavigationView.swift:17]: same question about this

File: Sources/SmileID/Classes/BiometricKYC/OrchestratedBiometricKycViewModel.swift [hunk 3]
```diff
@@ -38,7 +33,7 @@ class OrchestratedBiometricKycViewModel: ObservableObject {
         if selfieFile != nil {
             submitJob()
         } else {
-            updateStep(.selfie)
+            updateProcessingState(nil)
```
Comment (jumaallan) [inline Sources/SmileID/Classes/BiometricKYC/OrchestratedBiometricKycViewModel.swift:36]: We should maybe have a state that is not `nil`

Like use an actual name so it's even easier to go through the code and understand

File: Sources/SmileID/Classes/BiometricKYC/OrchestratedBiometricKycViewModel.swift [hunk 4]
```diff
@@ -103,7 +98,7 @@ class OrchestratedBiometricKycViewModel: ObservableObject {
 
         guard selfieFile != nil else {
             // Set step to .selfieCapture so that the Retry button goes back to this step
-            updateStep(.selfie)
+            updateProcessingState(nil)
```
Comment (jumaallan) [inline Sources/SmileID/Classes/BiometricKYC/OrchestratedBiometricKycViewModel.swift:101]: same here

File: Sources/SmileID/Classes/Navigation/CancellableNavigationView.swift [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+    var body: some View {
+        NavigationView {
+            content()
+                .navigationBarItems(
+                    leading: Button {
+                        onCancel()
+                    } label: {
+                        Text(SmileIDResourcesHelper.localizedString(for: "Action.Cancel"))
+                            .foregroundColor(SmileID.theme.accent)
```
Comment (jumaallan) [inline Sources/SmileID/Classes/Navigation/CancellableNavigationView.swift:23]: theme question here since this is inside the sdk

File: Sources/SmileID/Classes/SmileID.swift [hunk 6]
```diff
@@ -411,13 +411,16 @@ public class SmileID {
     /// - Parameters:
     ///  - config: configuration options for customising biometric verification experience
     ///  - delegate: Callback to be invoked when the Biometric KYC is complete.
+    ///  - onDismiss: Closure to provide custom implementation when biometric kyc screen is dismissed
     public class func biometricKycScreen(
         config: BiometricVerificationConfig,
-        delegate: BiometricKycResultDelegate
+        delegate: BiometricKycResultDelegate,
+        onDismiss: (() -> Void)? = nil
```
Comment (jumaallan) [inline Sources/SmileID/Classes/SmileID.swift:418]: we don't expose this on android - not opposed to it, but maybe we need to add something like that on android so that the wrappers are in sync

We also use the delegate to send the user cancelled action since you can use the back buttons on android

File: Sources/SmileID/Classes/BiometricKYC/OrchestratedBiometricKycViewModel.swift (hunk 3 above)
Comment (tobitech) [reply Sources/SmileID/Classes/BiometricKYC/OrchestratedBiometricKycViewModel.swift:36]: The state is used to drive navigation to the ProcessingScreen. nil means the job submission hasn't started, so processing screen isn't showing. That's why it's an optional value.

File: Sources/SmileID/Classes/SmileID.swift (hunk 6 above)
Comment (tobitech) [reply Sources/SmileID/Classes/SmileID.swift:418]: Yeah, basically this is an orchestrated flow. We discussed that we put a constraint on how we want orchestrated flows to be presented. The goal was for orchestrated flows to only be presented as an interstitials and can be dismissed so that users can return back to the original flow. 
This closure is for developers to handle that dismiss if they want to... notice that it's optional. 
The other goal here is to separate navigation actions from job submission events (didSucceed or didError).

File: Example/SmileID/BiometricKYC/BiometricKycWithIdInputScreen.swift (hunk 1 above)
Comment (tobitech) [reply Example/SmileID/BiometricKYC/BiometricKycWithIdInputScreen.swift:19]: This is from the theme class. Not affected by dark mode on the device. 
PS: This is the same as on the main branch. I only wrapped the body of the screen inside a ZStack to prevent glitches.

File: Sources/SmileID/Classes/SelfieCapture/View/OrchestratedEnhancedSelfieCaptureScreen.swift [hunk 7]
```diff
@@ -29,40 +29,45 @@ public struct OrchestratedEnhancedSelfieCaptureScreen: View {
...
+            ZStack {
+                if showInstructions {
+                    LivenessCaptureInstructionsView(
+                        showAttribution: config.showAttribution,
+                        didTapGetStarted: {
+                            showInstructions = false
+                        }
+                    )
+                    .transition(.move(edge: .leading))
+                } else {
```
[the remark this answers is not in the document]
Comment (tobitech) [reply Sources/SmileID/Classes/SelfieCapture/View/OrchestratedEnhancedSelfieCaptureScreen.swift:42]: It's currently determined by the state right now but the state needs an initial value that is based on the configuration supplied (for customisation). If showInstructions is true in config that will be the default value and it will be updated when instructions haven been read so that the capture screen can now be displayed.

File: Sources/SmileID/Classes/BiometricKYC/OrchestratedBiometricKycViewModel.swift (hunk 3 above)
Comment (robin-smileid) [reply Sources/SmileID/Classes/BiometricKYC/OrchestratedBiometricKycViewModel.swift:40]: I'd agree - we should have clear defined states. An optional seems a bit of a hack. Maybe we can call it `unSubmitted` or `notStarted`, or `init`?
